src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,386 @@
+            var gitHubToken = await gitHubTokenProvider.GetTokenForInstallationAsync(installationId);
+            Logger.LogInformation($"GitHub token acquired for '{issueRepo}'");
+            var version = Assembly.GetExecutingAssembly()
+                .GetCustomAttribute<AssemblyInformationalVersionAttribute>()
+                .InformationalVersion;
+            Octokit.ProductHeaderValue product = new Octokit.ProductHeaderValue("Maestro", version);
+            return new GitHubClient(product)
+            {
+                Credentials = new Credentials(gitHubToken),
+            };
+        }
+
+        private async Task CreateOrUpdateGithubIssueAsync(RepositoryBranchUpdateHistoryEntry updateHistoryError, string issueRepo)
+        {
+            Logger.LogInformation($"Error Message : '{updateHistoryError.ErrorMessage}' in repository :  '{updateHistoryError.Repository}'");
+            var gitHubIssueEvaluator =
+                await _stateManager.GetOrAddAsync<IReliableDictionary<(string, string), int>>("gitHubIssueEvaluator");
+            var client = await AuthenticateGitHubClient(issueRepo);
+            var repo = await client.Repository.Get(_options.Owner, _options.Repository);
+            var issueNumber = new ConditionalValue<int>();
+            using (ITransaction tx = _stateManager.CreateTransaction())
+            {
+                issueNumber = await gitHubIssueEvaluator.TryGetValueAsync(tx,
+                    (updateHistoryError.Repository, updateHistoryError.Branch));
+                await tx.CommitAsync();
+            }
+            if (issueNumber.HasValue)
+            {
+                Logger.LogInformation(
+                    $"Updating a gitHub issue number : '{issueNumber}' for the error : '{updateHistoryError.ErrorMessage}'"
+                    + $"for the repository : '{updateHistoryError.Repository}'");
+                var issue = await client.Issue.Get(repo.Id, issueNumber.Value);
+                var descriptionUpdateFlag = false;
+                var comments = await client.Issue.Comment.GetAllForIssue(repo.Id, issue.Number);
+                var updateIssueDescription = CreateOrUpdateDescription(updateHistoryError, issue.Body);
+                // check if the issue body has to be updated or comment body has to be updated.
+                if (updateIssueDescription.Item2)
+                {
+                    // Issue body will be updated
+                    descriptionUpdateFlag = true;
+                    var issueUpdate = issue.ToUpdate();
+                    var bodyTitle = $"The following errors have been detected when attempting to update dependencies in " +
+                        $"'{updateHistoryError.Repository}' \n\n";
+                    issueUpdate.Body = bodyTitle + updateIssueDescription.Item1 + "\n\n**/FyiHandle :** " + _options.FyiHandle;
+                    issueUpdate.State = ItemState.Open;
```

Comment:
What happens if the issue has been closed? are we reopening it? shouldn't we simply create a new issue if the previous one has been closed?

Reply:
if the issue is closed, I am reopening it. As I have a list of repo + branch + issue number in the reliable dictionary, I m not deleting the older combo. I think keeping the same issue is good. Creating a new issue, comes with a overhead of deleting the older one in the dictionary.

Reply:
Hmm, think about this scenario:

An issue is opened on 3/20/2020. We discover it's a problem with how the subscription is configured. We add a bunch of comments to the issue, describe the fix, and close it.

5 months later, the subscription is modified, and it gets misconfigured again in the same way. Would it make sense to reopen the issue that was closed 5 months ago and update the issue body?

Reply:
I changed the update part to check if the issue is closed then create a new issue